src/commands/ui.rs

```diff
@@ -10,37 +10,66 @@ use ring::{aead, agreement, digest, rand, signature};
 use crate::commands::ExitCode;
 use crate::options::{Options, UI};
 use crate::platform::data_dir;
-use crate::portable::local::{instance_data_dir, InstanceInfo};
+use crate::portable::local::{instance_data_dir, NonLocalInstance};
 use crate::print;
 
 pub fn show_ui(options: &Options, args: &UI) -> anyhow::Result<()> {
     let connector = options.create_connector()?;
     let builder = connector.get()?;
     let mut url = format!("http://{}:{}/ui", builder.get_host(), builder.get_port());
-    if let Some(instance) = builder.get_instance_name() {
-        if instance != "_localdev" {
-            let ver = InstanceInfo::read(instance)?.get_version()?.specific();
-            if ver.major < 2 {
+
+    if !args.no_server_check {
+        match open_url(&url).map(|r| r.status()) {
+            Ok(reqwest::StatusCode::OK) => (),
+            Ok(reqwest::StatusCode::NOT_FOUND) => {
+                print::error("the specified EdgeDB server is not serving Web UI.");
+                print::echo!(
+                    "  If you have EdgeDB 2.0 and above, try to run the \
+                    server with `--admin-ui=enabled`."
+                );
+                return Err(ExitCode::new(2).into());
+            }
+            Ok(_) => {
+                print::error(
+                    "the specified EdgeDB server is not serving Web UI \
+                    correctly; check server log for details.",
+                );
+                return Err(ExitCode::new(3).into());
+            }
+            Err(e) => {
                 print::error(format!(
-                    "the specified instance runs EdgeDB v{}, which does not \
-                    include the Web UI; consider upgrading the instance to \
-                    EdgeDB 2.x or later",
-                    ver,
+                    "cannot connect to the specified EdgeDB server: {:#}",
+                    e,
                 ));
-                return Err(ExitCode::new(2).into())
+                return Err(ExitCode::new(4).into());
```

Comment:
Should this be warning instead? I think this check is merely informative. There still are reasons why you could connect via browser but not via CLI (i.e. proxy only set up in browser).

Reply:
> proxy only set up in browser

In that case - quoting Elvis - the CLI should usually have the `HTTP_PROXY` in its env and `reqwest` is supposed to handle that correctly. I'm convinced that in a dev env, if the CLI cannot open UI, the browser is likely unable to.

Reply:
At least print URL besides the error, so if proxy is set up in the wrong way, there is something to debug.